app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -0,0 +1,9 @@
+module ManageIQ::Providers::Azure::CloudManager::Provision::StateMachine
+  def customize_destination
```

Comment:
@durandom Google does inherit from the cloud state machine but there is no selection for floating IP so `if floating_ip` will always be false and the `associate_floating_ip` method will not be called.

Now that you pointed it out this should probably be refactored to remove any chance of it being inadvertently called.